src/modules/expression/binop/range.rs

```diff
@@ -28,16 +48,18 @@ impl SyntaxModule<ParserMetadata> for Range {
     }
 
     fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
-        parse_left_expr(meta, self.from.as_mut(), "..")?;
-        let tok = meta.get_current_token();
-        token(meta, "..")?;
-        token(meta, "=").is_err().then(|| self.neq = true);
-        syntax(meta, self.to.as_mut())?;
-        let l_type = self.from.get_type();
-        let r_type = self.to.get_type();
-        let message = format!("Cannot create a range starting from value of type '{l_type}' up until value of type '{r_type}'");
-        let predicate = |kind| matches!(kind, Type::Num);
-        expression_arms_of_type(meta, &l_type, &r_type, predicate, tok, &message)?;
+        if self.from.get_type() != Type::Num {
+            let l_type = self.from.get_type();
+            let err = self.from.get_error_message(meta)
+                .message(format!("Range can only work on type 'Num', but '{l_type}' was provided"));
+            return Err(Failure::Loud(err));
+        }
+        if self.to.get_type() != Type::Num {
+            let r_type = self.to.get_type();
+            let err = self.to.get_error_message(meta)
+                .message(format!("Range can only work on type 'Num', but '{r_type}' was provided"));
```

Comment:
Isn't it just a binop with `..` operator, and `Num` as the only acceptable type?